Scanner implementation

KarimGuesmi opened the pull request:
Pull request creation of the Travel_Agency2
Scanner external class , interface and it's implementations

KarimGuesmi committed:
Scanner implementation

budaimartin commented on src/hu/elte/txtuml/stdlib/util/Scanner.java:
```diff
@@ -0,0 +1,76 @@
+	/*
+	Constructs a new Scanner that produces values scanned from the specified file.
+	 */
+	public void scanner(File file);
```
Do not use `java.io.File` because it is not a txtUML class nor an external class.

KarimGuesmi replied:
/*
It's done, i've changes the File into a String value which specifies the path of the file
*/
@Override
	public void Scanner(String source) {
			sc = new Scanner(source);
	}

KarimGuesmi committed:
deleting io.File and replacing by String of the path, add JAVADOC and finally adding radix parameter

budaimartin commented on src/hu/elte/txtuml/stdlib/util/ScannerImplementation.java:
```diff
@@ -0,0 +1,77 @@
+
+	@Override
+	public boolean hasNextInt(int radix) {
+		return sc.hasNextInt();
```
Why don't you use the `radix` parameter?

KarimGuesmi replied:
//It's Done .
public boolean hasNextInt(int radix) {
		return sc.hasNextInt(radix);
}

budaimartin reviewed (changes requested):
Sorry for the late response. There are some issues with your work. It doesn't have any Javadoc comments. For instruction see my comments here: https://github.com/ELTE-Soft/txtUML-stdlib/pull/8 . Also see my other comments below.

KarimGuesmi commented:
//For the JAVADOC i will create this :

package hu.elte.txtuml.stdlib.util;

import java.io.File;
import java.io.FileNotFoundException;

import hu.elte.txtuml.api.model.external.ExternalClass;

public interface Scanner extends ExternalClass{
	/**
     * Constructs a new <code>Scanner</code> that produces values scanned
     * from the specified string.
     *
     * @param  source A string to scan
     */
	public void Scanner(String source);
	 
	 
	 /**
     * Returns true if this scanner has another token in its input.
     * This method may block while waiting for input to scan.
     * The scanner does not advance past any input.
     *
     * @return true if and only if this scanner has another token
     * @throws IllegalStateException if this scanner is closed
     * @see java.util.Iterator
     */
	public boolean hasNext();
	
	
	/**
     * Returns true if the next token matches the pattern constructed from the
     * specified string. The scanner does not advance past any input.
     *
     * An invocation of this method of the form hasNext(pattern)
     * behaves in exactly the same way as the invocation
     * hasNext(Pattern.compile(pattern)).
     *
     * @param pattern a string specifying the pattern to scan
     * @return true if and only if this scanner has another token matching
     *         the specified pattern
     * @throws IllegalStateException if this scanner is closed
     */
	public boolean hasNext(String pattern);
	
	
	/**
     * Returns true if the next token in this scanner's input can be
     * interpreted as an int value in the specified radix using the
     * {@link #nextInt} method. The scanner does not advance past any input.
     *
     * @param radix the radix used to interpret the token as an int value
     * @return true if and only if this scanner's next token is a valid
     *         int value
     * @throws IllegalStateException if this scanner is closed
     */
	public boolean hasNextInt(int radix);
	
	
	
	/**
     * Returns true if the next token in this scanner's input can be
     * interpreted as an int value in the default radix using the
     * {@link #nextInt} method. The scanner does not advance past any input.
     *
     * @return true if and only if this scanner's next token is a valid
     *         int value
     * @throws IllegalStateException if this scanner is closed
     */
	public boolean hasNextInt();
	
	
	/**
     * Returns true if there is another line in the input of this scanner.
     * This method may block while waiting for input. The scanner does not
     * advance past any input.
     *
     * @return true if and only if this scanner has another line of input
     * @throws IllegalStateException if this scanner is closed
     */
	public boolean hasNextLine();
	
	
	/**
     * Finds and returns the next complete token from this scanner.
     * A complete token is preceded and followed by input that matches
     * the delimiter pattern. This method may block while waiting for input
     * to scan, even if a previous invocation of {@link #hasNext} returned
     * <code>true</code>.
     *
     * @return the next token
     * @throws NoSuchElementException if no more tokens are available
     * @throws IllegalStateException if this scanner is closed
     * @see java.util.Iterator
     */
	public String next();
	
	
	/**
     * Scans the next token of the input as an <tt>int</tt>.
     *
     *  An invocation of this method of the form
     * nextInt() behaves in exactly the same way as the
     * invocation nextInt(radix), whereradix
     * is the default radix of this scanner.
     *
     * @return the int scanned from the input
     * @throws InputMismatchException
     *         if the next token does not match the Integer
     *         regular expression, or is out of range
     * @throws NoSuchElementException if input is exhausted
     * @throws IllegalStateException if this scanner is closed
     */
	public int nextInt();
	
	
	/**
     * Scans the next token of the input as an <tt>int</tt>.
     * This method will throw <code>InputMismatchException</code>
     * if the next token cannot be translated into a valid int value as
     * described below. If the translation is successful, the scanner advances
     * past the input that matched.
     *
     *  If the next token matches the 
     * regular expression defined
     * above then the token is converted into an int value as if by
     * removing all locale specific prefixes, group separators, and locale
     * specific suffixes, then mapping non-ASCII digits into ASCII
     * digits via {@link Character#digit Character.digit}, prepending a
     * negative sign (-) if the locale specific negative prefixes and suffixes
     * were present, and passing the resulting string to
     * {@link Integer#parseInt(String, int) Integer.parseInt} with the
     * specified radix.
     *
     * @param radix the radix used to interpret the token as an int value
     * @return the <tt>int</tt> scanned from the input
     * @throws InputMismatchException
     *         if the next token does not match the <i>Integer</i>
     *         regular expression, or is out of range
     * @throws NoSuchElementException if input is exhausted
     * @throws IllegalStateException if this scanner is closed
     */
	public int nextInt(int radix);
	
	
	
	/**
     * Advances this scanner past the current line and returns the input
     * that was skipped.
     *
     * This method returns the rest of the current line, excluding any line
     * separator at the end. The position is set to the beginning of the next
     * line.
     *
     * Since this method continues to search through the input looking
     * for a line separator, it may buffer all of the input searching for
     * the line to skip if no line separators are present.
     *
     * @return the line that was skipped
     * @throws NoSuchElementException if no line was found
     * @throws IllegalStateException if this scanner is closed
     */
	public String nextLine();
	
	
	/**
     * Returns the string representation of this Scanner. The
     * string representation of a Scanner contains information
     * that may be useful for debugging. The exact format is unspecified.
     *
     * @return  The string representation of this scanner
     */
	public String toString();
	
	
	/**
     * The remove operation is not supported by this implementation of
     * Iterator.
     *
     * @throws UnsupportedOperationException if this method is invoked.
     * @see java.util.Iterator
     */
	public void remove();
	
	/**
     * Resets this scanner.
     *
     *  Resetting a scanner discards all of its explicit state
     * information which may have been changed by invocations of {@link
     * #useDelimiter}, {@link #useLocale}, or {@link #useRadix}.
     *
     *  An invocation of this method of the form
     * scanner.reset()</tt> behaves in exactly the same way as the
     * invocation
     *
     * 
     *   scanner.useDelimiter("\\p{javaWhitespace}+")
     *          .useLocale(Locale.getDefault(Locale.Category.FORMAT))
     *          .useRadix(10);
     * 
     *
     * @return this scanner
     *
     * @since 1.6
     */
	public void reset();
	
	
	
	/**
     * Attempts to find the next occurrence of a pattern constructed from the
     * specified string, ignoring delimiters.
     *
     * An invocation of this method of the form findInLine(pattern)
     * behaves in exactly the same way as the invocation
     * findInLine(Pattern.compile(pattern))
     *
     * @param pattern a string specifying the pattern to search for
     * @return the text that matched the specified pattern
     * @throws IllegalStateException if this scanner is closed
     */
	public String findInLine(String pattern);
	
}

KarimGuesmi commented:
Dear Mr.  Budai Martin,
Thank you for your advices, i made a review for the code now, - I replaced the IO.FILE by a String parameter which represents the path of the FILE, it's a better option i think to use String instead of external FILE.- I added the radix parameter .- I added the javaDoc comments to the interface Scanner.
Best Regards,Karim GUESMI

      De : Budai Martin <notifications@github.com>
 À : ELTE-Soft/txtUML-stdlib <txtUML-stdlib@noreply.github.com> 
Cc : KarimGuesmi <karim_poste@yahoo.fr>; Author <author@noreply.github.com>
 Envoyé le : Mercredi 8 février 2017 23h14
 Objet : Re: [ELTE-Soft/txtUML-stdlib] Scanner implementation (#11)
   
@bmartin0121 requested changes on this pull request.Sorry for the late response. There are some issues with your work. It doesn't have any Javadoc comments. For instruction see my comments here: #8 . Also see my other comments below.In src/hu/elte/txtuml/stdlib/util/Scanner.java:> @@ -0,0 +1,76 @@
+package hu.elte.txtuml.stdlib.util;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+
+import hu.elte.txtuml.api.model.external.ExternalClass;
+
+public interface Scanner extends ExternalClass{
+	/*
+	Constructs a new Scanner that produces values scanned from the specified file.
+	 */
+	public void scanner(File file);
Do not use java.io.File because it is not a txtUML class nor an external class.In src/hu/elte/txtuml/stdlib/util/ScannerImplementation.java:> +		}
+	}
+
+	@Override
+	public boolean hasNext() {
+		return sc.hasNext();
+	}
+
+	@Override
+	public boolean hasNext(String pattern) {
+		return sc.hasNext(pattern);
+	}
+
+	@Override
+	public boolean hasNextInt(int radix) {
+		return sc.hasNextInt();
Why don't you use the radix parameter?—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.
